### PR TITLE
Add validations to SiteConfig model

### PR DESCRIPTION
**Pull request by citizen428**
## What type of PR is this? (check all applicable)

- [X] Refactor

## Description

Until recently the [`rails_settings_cached` gem](https://github.com/huacnlee/rails-settings-cached) did not support validations, so we performed them explicitly, first in `Admin::ConfigsController` and later in the `SiteConfigs::Upsert` service object. 

This PR moves validations into the model itself. Note that the gem only supports the `validates` option, but not custom validation methods  `validate`, so I had to add a few new validator classes to `app/validators`.

## Related Tickets & Documents

#11991 

## QA Instructions, Screenshots, Recordings

Screenshots of the different validation errors:

image
image
image
image

Note: the hex color validation is hard to trigger from the UI as we are using a color input.

For QAing this, go to `/admin/config` and try setting invalid and valid values respectively.

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://docs.forem.com/frontend/accessibility)._

## Added tests?

- [X] Yes
- 
## Added to documentation?

- [X] No documentation needed

### CODE REVIEW

File: app/models/site_config.rb [hunk 1]
```diff
@@ -4,14 +4,15 @@
 class SiteConfig < RailsSettings::Base
   self.table_name = "site_configs"
 
-  validates :var, presence: true
```
Comment (citizen428) [inline app/models/site_config.rb:7]: This has me stumped a bit. While this line works perfectly fine on `master` it breaks on this branch, even though the gem version is the same and no relevant code was changed.

Master:
image

This branch:
image

The column itself already has a non-null constraint, so it's not possible to insert an invalid value. We also never directly interact with `SiteConfig` records but only do so through the gem's interface. 

I disabled this in order to not get stuck in a debugging rabbit hole for a validation that only seems marginally useful to me, but I'm happy to dig deeper if people feel strongly about it.

File: app/validators/color_contrast_validator.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class ColorContrastValidator < ActiveModel::EachValidator
```
Comment (rhymes) [inline app/validators/color_contrast_validator.rb:1]: ❤️

**Review by rhymes — APPROVED** [review]
Wow! So happy they added this and thanks for the great refactoring.

Let's get this merged :D

File: app/models/site_config.rb (hunk 1 above)
Comment (citizen428) [reply app/models/site_config.rb:7]: For the record, @rhymes and I discussed this on Slack and agreed that this validation can be safely removed.

**Review by Ridhwana — APPROVED** [review]
This looks so much better! 😍

**Pull request merged**